packages/contracts-bedrock/src/periphery/drippie/dripchecks/CheckSecrets.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { IDripCheck } from "../IDripCheck.sol";
+
+/// @title CheckSecrets
+/// @notice DripCheck that checks if specific secrets exist (or not). Supports having a secret that
+///         must exist for the check to pass as well as a second secret that must not exist. First
+///         secret can be revealed to begin the drip, second secret can be revealed to stop it.
+contract CheckSecrets is IDripCheck {
+    struct Params {
+        uint256 delay;
+        bytes32 secretMustExist;
+        bytes32 secretMustNotExist;
+    }
+
+    /// @notice External event used to help client-side tooling encode parameters.
+    /// @param params Parameters to encode.
+    event _EventToExposeStructInABI__Params(Params params);
+
+    /// @notice Event emitted when a secret is revealed.
+    event SecretRevealed(bytes32 secret);
```

Comment:
would it be beneficial to index this arg, and also emit the unhashed secret here?
```suggestion
    event SecretRevealed(bytes32 indexed secretHash, bytes secret);
```

Reply:
Yeah sure, why not. I'll add it.